vyperlang/vyper
# feat: add support for modules with variables

### What I did
add support for modules with variables. modules are instantiated separately from import statements, decoupling code and storage layout concerns.

### How I did it
- change the allocator to handle modules as variables
- add analysis support for touched locations
- change the calling convention to include pointers to storage/immutables/whatever

note that this feature probably requires dropping support for storage layout overrides (at least in its current form)

### How to verify it
still incomplete. need to make sure support for immutables is correct, and add tests.
- unit tests for the analysis
- various combinations of importing modules with storage and/or immutables
- tests for the storage allocator
- handling of `__init__()` functions

### Commit message

Commit message for the final, squashed PR. (Optional, but reviewers will appreciate it! Please see [our commit message style guide](../../master/docs/style-guide.rst#best-practices-1) for what we would ideally like to see in a commit message.)

### Description for the changelog

### Cute Animal Picture

![Put a link to a cute animal picture inside the parenthesis-->]()
